pulp_smash/pulp3/bindings.py

```diff
@@ -59,7 +68,34 @@ def monitor_task(task_href):
         sleep(SLEEP_TIME)
         task = tasks.read(task_href)
 
-    if task.state == "completed":
-        return task.created_resources
+    if task.state != "completed":
+        raise PulpTaskError(task=task)
```

Comment:
While you are at it, should we add a timeout?

Reply:
I don't think we can set a default timeout since this function is used in tons of places in very different situations, but I'm not against adding the feature itself.  Do we have a specific use case?

Reply:
If we come up with one, let's just add it separately.  Want to go ahead and get this merged so I can close out a dozen tabs :)

Reply:
I was thinking of a hanging task. It would be nicer if the CI didn't exit with "Your time is up".

Reply:
@mdellweg The problem is, it's hard to find a good generic value for how long to wait, so it'd need to be pretty high by default, or be off by default.  If it's off by default, we'd need to go through quite some effort to use the feature.  If it's on by default, then all long-running tasks (like the performance tests) need to explicitly disable it, which seems kind of annoying.

Reply:
It's probably still useful I just don't want to worry about it right now.  It could require many changes across repos again.